### PR TITLE
Fix SSL errors after forking process

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: src/neptune/metadata_containers/metadata_container.py [hunk 1]
```diff
@@ -143,6 +147,43 @@ def __init__(
...
+        if sys.version_info >= (3, 7):
+            try:
+                os.register_at_fork(
+                    after_in_child=self._handle_fork_in_child, after_in_parent=self._handle_fork_in_parent
+                )
+            except AttributeError:
+                pass
+
+    """
+    OpenSSL's internal random number generator does not properly handle forked processes.
```
Comment (Raalsky) [inline src/neptune/metadata_containers/metadata_container.py:159]: Let's move the comment and `ssl.RAND_bytes(100)` to a separate utility function.

File: src/neptune/internal/operation_processors/async_operation_processor.py [hunk 2]
```diff
@@ -73,35 +72,25 @@ def __init__(
...
-                os.register_at_fork(after_in_child=self._handle_fork_in_child)
-            except AttributeError:
-                pass
-
     @staticmethod
     def _init_data_path(container_id: UniqueId, container_type: ContainerType):
         now = datetime.now()
         container_dir = f"{NEPTUNE_DATA_DIRECTORY}/{ASYNC_DIRECTORY}/{container_type.create_dir_name(container_id)}"
-        data_path = f"{container_dir}/exec-{now.timestamp()}-{now.strftime('%Y-%m-%d_%H.%M.%S.%f')}"
+        data_path = f"{container_dir}/exec-{os.getpid()}-{now.timestamp()}-{now.strftime('%Y-%m-%d_%H.%M.%S.%f')}"
```
Comment (Raalsky) [inline src/neptune/internal/operation_processors/async_operation_processor.py:89]: Let's move the process id to the end. Currently, it breaks "sorting" capabilities :D.

File: src/neptune/internal/operation_processors/async_operation_processor.py [hunk 3]
```diff
@@ -73,35 +72,25 @@ def __init__(
         self._container_id = container_id
         self._container_type = container_type
         self._backend = backend
+        self._lock = lock
```
Comment (Raalsky) [inline src/neptune/internal/operation_processors/async_operation_processor.py:75]: No need to store it (and `sleep_time` as well)

**Comment by PatrykGala** [comment]
No need to register at fork in HostedNeptuneBackendApiClient and HostedAlphaLeaderboardApiClient.
```      
  if sys.version_info >= (3, 7):
            try:
                os.register_at_fork(after_in_child=self._handle_fork_in_child)
            except AttributeError:
                pass

    def _handle_fork_in_child(self):
        self.backend_swagger_client = NoopObject()

```

**Comment by aniezurawski** [comment]
> No need to register at fork in HostedNeptuneBackendApiClient and HostedAlphaLeaderboardApiClient.

@PatrykGala it's a legacy API. I don't want to touch its behaviour without a good reason.

File: src/neptune/metadata_containers/metadata_container.py [hunk 4]
```diff
@@ -143,6 +147,43 @@ def __init__(
 
         self._startup(debug_mode=mode == Mode.DEBUG)
 
+        if sys.version_info >= (3, 7):
```
[the remark this answers is not in the document]
Comment (aniezurawski) [reply src/neptune/metadata_containers/metadata_container.py:150]: Done.

File: src/neptune/metadata_containers/metadata_container.py (hunk 1 above)
Comment (aniezurawski) [reply src/neptune/metadata_containers/metadata_container.py:159]: Done.

File: src/neptune/internal/operation_processors/async_operation_processor.py (hunk 2 above)
Comment (aniezurawski) [reply src/neptune/internal/operation_processors/async_operation_processor.py:89]: Done.

File: src/neptune/internal/operation_processors/async_operation_processor.py (hunk 3 above)
Comment (aniezurawski) [reply src/neptune/internal/operation_processors/async_operation_processor.py:75]: Done.

File: src/neptune/metadata_containers/metadata_container.py [hunk 5]
```diff
@@ -143,6 +148,37 @@ def __init__(
 
         self._startup(debug_mode=mode == Mode.DEBUG)
 
+        os.register_at_fork(after_in_child=self._handle_fork_in_child, after_in_parent=self._handle_fork_in_parent)
```
Comment (Raalsky) [inline src/neptune/metadata_containers/metadata_container.py:151]: You can use `reset_internal_ssl_state` directly here (IDE suggestion)

File: src/neptune/metadata_containers/metadata_container.py [hunk 6]
```diff
@@ -143,6 +148,37 @@ def __init__(
...
+
+    """
```
Comment (Raalsky) [inline src/neptune/metadata_containers/metadata_container.py:153]: Can we move this comment to `reset_internal_ssl_state` as well?

File: src/neptune/metadata_containers/metadata_container.py (hunk 5 above)
Comment (aniezurawski) [reply src/neptune/metadata_containers/metadata_container.py:151]: I don't like this suggestion. :)

File: src/neptune/metadata_containers/metadata_container.py (hunk 6 above)
Comment (aniezurawski) [reply src/neptune/metadata_containers/metadata_container.py:153]: Done.

**Pull request merged**